web/src/components/search.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { Locale, useLocale } from "./locale";
+import { useSearch } from "src/utils/search";
+import { useSearchModal } from "src/utils/search-modal";
+import { ContributionCard } from "./contribution-card";
+import { ContributorCard } from "./contributor-card";
+import { ProjectCard } from "./project-card";
+import { GetContributionsResponse } from "@dzcode.io/api/dist/contribution/types";
+import { GetContributorsResponse } from "@dzcode.io/api/dist/contributor/types";
+import { GetProjectsResponse } from "@dzcode.io/api/dist/project/types";
+
+export function Search(): JSX.Element {
+  const { localize } = useLocale();
+  const [query, setQuery] = useState("");
+
+  const { results, isFetching } = useSearch(query);
+
+  const { hideModal, showModal } = useSearchModal();
+
+  const onKeyDown = useCallback(
+    (event: KeyboardEvent) => {
+      if (event.key === "/") {
+        event.preventDefault();
+        showModal();
+      }
+    },
+    [showModal],
+  );
+
+  useEffect(() => {
+    document.addEventListener("keydown", onKeyDown);
+    return () => {
+      document.removeEventListener("keydown", onKeyDown);
+    };
+  }, [onKeyDown]);
```

Comment:
can be simplified: to:

```suggestion
  useEffect(() => {
    const onKeyDown = (event: KeyboardEvent) => {
      if (event.key === "/") {
        event.preventDefault();
        showModal();
      }
    };

    document.addEventListener("keydown", onKeyDown);
    return () => {
      document.removeEventListener("keydown", onKeyDown);
    };
  }, [showModal]);
```